### DISCUSSION TITLE
Nuxt 3 page component name not showing up in Vue DevTools

### DISCUSSION THREAD

**Discussion by nathanchase** [nuxt/nuxt#17254] [Ideas]
I noticed that all Nuxt 3 page components show up as "Index" in Vue DevTools (unless they're dynamic with a slug, or have a filename other than `index.vue`).

Is there a way we can hint to Vue DevTools that the page component at `/pages/user/index.vue` should be called "User" and not "Index"?

The only workaround I've found so far is to include a second script tag using Options API to name the page:
```
<script>
export default {
  name: 'User',
};
</script>
```
I tried using `defineComponent` - but that doesn't work (DevTools still shows "Index"):
```
<script setup lang="ts">
defineComponent({
  name: 'User'
})
</script>
```
Less ideal behavior:
![image](https://user-images.githubusercontent.com/5963574/179752543-2be8aad6-6fc3-4722-a024-a4f86c713e40.png)
Desired behavior:
![image](https://user-images.githubusercontent.com/5963574/179752512-c459a76a-e8ab-4f41-b6b5-389d7008c38d.png)

**Comment by danielroe** [comment]
That's the correct approach: https://vuejs.org/api/sfc-script-setup.html#usage-alongside-normal-script. Vue does not currently support defining a component name within `<script setup>`

**Reply by nathanchase** [reply]
Ok!

**Reply by danielroe** [reply]
FYI, vue does _now_ support `defineOptions` in script setup for defining component names.

**Comment by m-shum** [comment]
Any news on this? Still having this issue in 2024 - it also prevents the component selector tool from working correctly when directory structure is used to namespace components.

**Reply by danielroe** [reply]
Would you create an issue in that case, with a reproduction? 🙏
